lib/rules/camelcase.js

```diff
@@ -113,24 +137,28 @@ module.exports = {
                 } else if (node.parent.type === "Property" || node.parent.type === "AssignmentPattern") {
 
                     if (node.parent.parent && node.parent.parent.type === "ObjectPattern") {
-
                         if (node.parent.shorthand && node.parent.value.left && isUnderscored(name)) {
 
                             report(node);
                         }
 
+                        const assignmentKeyEqualsValue = node.parent.key.name === node.parent.value.name;
```

Comment:
hard to see in the GitHub ui since I clobbered the old commit, but in the previous rev this was:
`const assignmentKeyEqualsValue = node.parent.key === node.parent.value;`
which failed the `{category_id: category_id} = query` test